scripts/custom-runtime.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+# This script  sets up the Rube Goldberg machine for testing against custom
+# revision of go-ipfs and js-ipfs (the latter being tricky due to lerna monorepo)
+#
+# It assumes IPFS_GO_EXEC or IPFS_JS_EXEC to be in env.
+# One can trigger this locally by exporting the same path as we do on CI.
+# For example, to run pubsub tests against go-ipfs and js-ipfs revision defined
+# in ./scripts/custom-runtime.sh one can:
+#
+#  export IPFS_GO_EXEC=/tmp/go-ipfs/cmd/ipfs/ipfs
+#  export IPFS_JS_EXEC=/tmp/js-ipfs/packages/ipfs/src/cli.js
+#  export IPFS_JS_MODULE=/tmp/js-ipfs/packages/ipfs/dist/cjs/src/index.js
+#  export IPFS_JS_HTTP_MODULE=/tmp/js-ipfs/packages/ipfs-http-client/dist/cjs/src/index.js
+#  ./scripts/custom-runtime.sh
+#  node bin/ipfs-interop.js -- -t node --grep "pubsub"
+
+set -eo pipefail
+
+WORKDIR=$(pwd)
+
+if [ "$IPFS_GO_EXEC" == /tmp/go-ipfs/cmd/ipfs/ipfs ]; then
+if [ ! -d /tmp/go-ipfs ]; then
+    cd /tmp
+        git clone https://github.com/coryschwartz/go-ipfs.git
+        cd go-ipfs
+            # implementation from https://github.com/ipfs/go-ipfs/pull/8183
+            git checkout 153697d524f449ee9bec97245b0fcd7ebc2e8170
```

Comment:
Switching to master in prep for https://github.com/ipfs/go-ipfs/pull/8183 to land.
```suggestion
        git clone https://github.com/ipfs/go-ipfs.git
        cd go-ipfs
            # implementation from https://github.com/ipfs/go-ipfs/pull/8183
            git switch master
```